btree/btree.c

```diff
@@ -6685,11 +6687,15 @@ void get_rec_at_index(struct nd *node, int idx, uint64_t *key,  uint64_t *val)
 
 	node_rec(&slot);
 
-	if (key != NULL)
-		*key = *(uint64_t *)p_key;
+	if (key != NULL) {
+		key[0] = ((uint64_t *)p_key)[0];
+		key[1] = ((uint64_t *)p_key)[1];
```

Comment:
here it is assumed that function will get pointer to array with array size is atleast 2, but for some of the ut's it will nt be the case and the functions will fail, This will work fine for btree_ut_kv_oper but not for all functions.

Moreover, ut_node_add_del_rec is using this function , but it is not comparing key[1] it is comparing key[0] which will be i guess keysize

Reply:
I had added these lines just for debugging purpose. Reverting these changes.